### PR TITLE
Improved compatibility of output tests with Python 3 #1951

### PR DESCRIPTION
**Pull request by Onager**
Improved compatibility of output tests with Python 3 #1951

### CODE REVIEW

File: tests/serializer/json_serializer.py [hunk 1]
```diff
@@ -149,6 +149,7 @@ def testReadAndWriteSerializedEventObject(self):
     expected_event_object.empty_string = ''
     expected_event_object.zero_integer = 0
     expected_event_object.integer = 34
+    expected_event_object.float = -122.082203542683
```
Comment (joachimmetz) [inline tests/serializer/json_serializer.py]: maybe rename event_object to event as well

File: tests/cli/test_lib.py [hunk 2]
```diff
@@ -55,6 +56,19 @@ def ReadOutput(self):
 
     return output_data
 
+  def ReadOutputString(self):
```
Comment (joachimmetz) [inline tests/cli/test_lib.py]: Why would you compare against string when what the original intent was to compare against the resulting encoded data.

File: tests/output/kml.py [hunk 3]
```diff
@@ -28,35 +28,35 @@ def setUp(self):
...
+    footer = self._output_writer.ReadOutputString()
     self.assertEqual(footer, expected_footer)
 
   def testWriteEventBody(self):
     """Tests the WriteEventBody function."""
 
     # Test event object without geo-location.
     self._output_module.WriteEventBody(self._event_object)
-    self.assertEqual(self._output_writer.ReadOutput(), b'')
+    self.assertEqual(self._output_writer.ReadOutputString(), '')
```
Comment (joachimmetz) [inline tests/output/kml.py]: suggestion: maybe break this up in
```
event_body = self._output_writer.ReadOutputString()
self.assertEqual(event_body, '')
```

so that it is more clear that we are testing event body

Comment (Onager) [reply tests/output/kml.py]: Done

File: tests/serializer/json_serializer.py (hunk 1 above)
Comment (Onager) [reply tests/serializer/json_serializer.py]: Done

File: tests/cli/test_lib.py (hunk 2 above)
Comment (Onager) [reply tests/cli/test_lib.py]: This class has a Write() method that takes a string, so it seems to me that this class deals with strings, not bytes. 

I don't think it's desirable to have output tests being dependent on the implementation of the writer class.

Test should need to care that the strings they expect are the ones they get, not how the strings are encoded when output. That sort of testing belongs in test for the writers themselves.

Comment (joachimmetz) [reply tests/cli/test_lib.py]: in that case maybe we should implement a test string output writer instead of encoding and decoding back and forth?

Comment (Onager) [reply tests/cli/test_lib.py]: Opened #1963 to track that

File: tests/cli/test_lib.py [hunk 4]
```diff
@@ -55,6 +56,20 @@ def ReadOutput(self):
 
     return output_data
 
+  # TODO: refactor this method https://github.com/log2timeline/plaso/issues/1963
```
Comment (joachimmetz) [inline tests/cli/test_lib.py]: On a second pass the intent of this class is to write output including encoding seeing some output format might specify a specific encoding. So that output is consistent cross-platform. Please undo these changes, seeing the no longer reflect what needs to be tested.

Comment (Onager) [reply tests/cli/test_lib.py]: Per conversation, I've changed the TestOutputWriter to just deal with strings, with no conversion between outputs and test fixtures (expected output).
